Source/objects.cpp

```diff
@@ -2349,24 +2349,25 @@ void OperateShrineHidden(Player &player)
 		while (true) {
 			cnt = 0;
 			for (auto &item : player.InvBody) {
-				if (!item.isEmpty() && item._iMaxDur != DUR_INDESTRUCTIBLE && item._iMaxDur != 0) {
+				if (!item.isEmpty() && HasNoneOf(item._iFlags, ItemSpecialEffect::Indestructible) && item._iMaxDur != 0) {
 					cnt++;
 				}
 			}
 			if (cnt == 0)
 				break;
 			int r = GenerateRnd(NUM_INVLOC);
-			if (player.InvBody[r].isEmpty() || player.InvBody[r]._iMaxDur == DUR_INDESTRUCTIBLE || player.InvBody[r]._iMaxDur == 0)
+			if (player.InvBody[r].isEmpty() || HasAnyOf(player.InvBody[r]._iFlags, ItemSpecialEffect::Indestructible) || player.InvBody[r]._iMaxDur == 0)
 				continue;
 
 			player.InvBody[r]._iDurability -= 20;
 			player.InvBody[r]._iMaxDur -= 20;
-			if (player.InvBody[r]._iDurability <= 0)
-				player.InvBody[r]._iDurability = 1;
-			if (player.InvBody[r]._iMaxDur <= 0)
-				player.InvBody[r]._iMaxDur = 1;
+
 			break;
 		}
+		for (auto &item : player.InvBody) {
+			item._iDurability = clamp(item._iDurability, 1, 255);

```

Comment:
```suggestion
			item._iDurability = clamp(item._iDurability, 1, 255);
                        item._iDurability = clamp(item._iDurability, 1, DUR_MAX);
```

Reply:
Could have sworn I committed this exact thing before getting off for the night lol. Guess I forgot to push